pkg/proxy/proxier.go

```diff
@@ -548,12 +555,33 @@ func (proxier *Proxier) openPortal(service types.NamespacedName, info *serviceIn
 	if err != nil {
 		return err
 	}
+
+	publicIpCount := 0
 	for _, publicIP := range info.publicIP {
-		err = proxier.openOnePortal(net.ParseIP(publicIP), info.portalPort, info.protocol, proxier.listenIP, info.proxyPort, service)
+		ip := net.ParseIP(publicIP)
+		if ip == nil {
+			glog.V(2).Info("Skipping non-address public endpoint: %s", publicIP)
+			continue
+		}
+		publicIpCount++
+		err = proxier.openOnePortal(ip, info.portalPort, info.protocol, proxier.listenIP, info.proxyPort, service)
 		if err != nil {
 			return err
 		}
 	}
+
+	// We have an external load balancer, but no specific IPs.
```

Comment:
This heuristic makes me very nervous.  It feels very loosely assembled with a bunch of little exception cases.

Did you look at the work on Rackspace LB's to switch the uiptables logic to recognize packets FROM the LB's IP?  Does that work for ELB?

I don't understand the DNS thing - are they just in the publicIPs list for completeness?